tests/skill_test.cpp

```diff
@@ -0,0 +1,43 @@
+#include "calendar.h"
+#include "cata_catch.h"
+#include "character.h"
+#include "player_helpers.h"
+#include "skill.h"
+
+TEST_CASE( "skill_rust_occurs", "[character][skill]" )
+{
+    Character &guy = get_player_character();
+    clear_avatar();
+
+    // Set every skill to two, so they can rust
+    for( const Skill &skill : Skill::skills ) {
+        INFO( skill.ident().str() );
+        guy.set_skill_level( skill.ident(), 2 );
+        CHECK( guy.get_skill_level( skill.ident() ) == 2.f );
+        CHECK( guy.get_knowledge_level( skill.ident() ) == 2 );
```

Comment:
Use `REQUIRE` in the setup so the test halts early.
```suggestion
        REQUIRE( guy.get_skill_level( skill.ident() ) == 2.f );
        REQUIRE( guy.get_knowledge_level( skill.ident() ) == 2 );
```
Also test `isRusty()` before and after.